In [4]:
import numpy as np
import pandas

## Problem 1

In [4]:
NNbird = [45,3]
NNperson = [5, 47]

NAbird = [47,1]
NAperson = [11,42]

<table>
  <thead>
    <tr>
      <th> </th>
      <th>True Bird</th>
      <th>True Person</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>NN Bird</th>
      <td>45</td>
      <td>5</td>
    </tr>
    <tr>
      <th>NN Person</th>  
      <td>3</td>
      <td>47</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th> </th>
      <th>True Bird</th>
      <th>True Person</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>NA Bird</th>
      <td>47</td>
      <td>10</td>
    </tr>
    <tr>
      <th>NA Person</th>
      <td>1</td>
      <td>42</td>
    </tr>
  </tbody>
</table>

1a) first makes less errors, the amount of errors only amounts to 8. The second one has 12 errors.

1b) 2nd better for zoo because it makes less errors about birds falsely reported as humans, and the zoo prefers to idenitify birds more accurately.

1c) Still one hundred photos, now only 1 true human and 99 true birds. mistaken bird = %100 that are left

In [4]:
NA_error_person = 42/52
NA_error_bird = 47/48

NN_error_person = 47/52
NN_error_bird = 45/48

NNbird2 = []
NNperson2 = []
NAbird2 = []
NAperson2 = []
NNbird2.append(594*NN_error_bird)
NNbird2.append(594*(1-NN_error_bird)) 

NNperson2.append(6*NN_error_person)
NNperson2.append(6*(1-NN_error_person))

NAbird2.append(594*NA_error_bird)
NAbird2.append(594*(1-NA_error_bird)) 

NAperson2.append(6*NA_error_person)
NAperson2.append(6*(1-NA_error_person))
print(NNbird2)
print(NNperson2)
print(NAbird2)
print(NAperson2)
print('Rounding the numbers to the nearest integer after multiplying the total amount of real birds/people by the previous algorithms % accuracy:')

[556.875, 37.125]
[5.423076923076923, 0.5769230769230769]
[581.625, 12.375000000000021]
[4.846153846153847, 1.1538461538461537]
Rounding the numbers to the nearest integer after multiplying the total amount of real birds/people by the previous algorithms % accuracy:


<table>
  <thead>
    <tr>
      <th> </th>
      <th>True Bird</th>
      <th>True Person</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>NN Bird</th>
      <td>557</td>
      <td>1</td>
    </tr>
    <tr>
      <th>NN Person</th>  
      <td>37</td>
      <td>5</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th> </th>
      <th>True Bird</th>
      <th>True Person</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>NA Bird</th>
      <td>582</td>
      <td>2</td>
    </tr>
    <tr>
      <th>NA Person</th>
      <td>12</td>
      <td>4</td>
    </tr>
  </tbody>
</table>

I chose a new number of photos, 600, so that we could see how the algorithm works where 1% of photos = 6. If we stuck with 100 photos then we would have both algorithms always making the right distinction for the one person, which doesn't help us understand the accuracy of the algorithm. Or at least we would get a non integer value for true and false sightings of the person, which does not make sense. 

## Problem 2

2a) PDF of 2 exponential distributions with equal parameters is an Erlang distribution with shape k = 2 and positive, real $\lambda$ parameter. Looking up the Erlang distribution, the Erlang distribution is just a special case of a gamma distribution where $\alpha = k$ and k is any positive integer. 

https://en.wikipedia.org/wiki/Exponential_distribution

https://en.wikipedia.org/wiki/Erlang_distribution

https://en.wikipedia.org/wiki/Gamma_distribution

2b) PDF of ratio of two zero-mean unity variance normal distributions is a standard cauchy distribution of mean = 0 and $\gamma$ = 0

https://en.wikipedia.org/wiki/Normal_distribution

https://en.wikipedia.org/wiki/Cauchy_distribution

2c) The PDF of the probability of vector amplitudes of a 2D Gaussian would be a Rayleigh distribution. A two-dimensional vector that has bivariate normally distributed components results in a Rayleigh distribution when you calculate the length of the vector as a probability density function.

https://en.wikipedia.org/wiki/Multivariate_normal_distribution

https://en.wikipedia.org/wiki/Rayleigh_distribution